marcelo_zoccoler/entry_user_interf3/Readme.md

```diff
@@ -0,0 +1,364 @@
+The [previous entry](https://biapol.github.io/blog/johannes_mueller/entry_user_interf2#creating-advanced-standalone-guis) showed you how to create more advanced GUIs with the aid of the designer tool. This third part will teach you how to import those GUIs to napari and how to produce them straight from python functions using [magicgui](https://napari.org/magicgui/index.html).
+
+## Table of contents
+* [Installing and running napari](#installing-and-running-napari)
+* [Calling napari from script](#calling-napari-from-script)
+* [Importing your fancy GUI to napari](#importing-your-fancy-gui-to-napari)
+* [Creating a callback function](#creating-a-callback-function)
+* [Automatically creating a GUI from a function with magicgui](#automatically-creating-a-gui-from-a-function-with-magicgui)
+* [Creating a GUI from FunctionGui](#creating-a-gui-from-functiongui)
+* Conclusion(#conclusion)
+
+## Installing and running napari
+
+The [napari website](https://napari.org/) has a more complete [installation tutorial](https://napari.org/tutorials/fundamentals/installation.html), but, overall, you should [create a new conda environment](https://biapol.github.io/blog/johannes_mueller/entry_user_interf2/Readme.md#creating-your-environment) and then type
+
+`conda install -c conda-forge napari`
+
+or
+
+`pip install napari[all]`
+
+in the command line. You can check if the installation was successful by calling `napari` from the command line and verifying if this window opens:
+
+![](images/napari_window.png)
+
+You can now add images to it by drag and drop! Pretty easy right? Why don't you give it a try before we continue?
+Download the image above and drop it into napari by drag and drop (or use the usual "File -> Open File(s)..."). You should have this:
+
+![](images/napari_window_in_napari.png)
+
+**Wow! We have napari inside napari!** ~~And that's how you add a GUI to napari! 😆~~
+
+## Calling napari from a script
+
+Sometimes, it may be convenient to call the napari viewer from a script in order to display a series of image processing steps pre-defined in Python code. This can be done in a couple lines:
+
+```
+import napari
+viewer = napari.Viewer()
+```
+
+The code above can be called from [Jupyter Notebook or JupyterLab](https://jupyter.org/), [Spyder](https://www.spyder-ide.org/), or your editor of preference.
+It is also possible to add images to the viewer from code. Download this image: [Astronaut photo of Tabuaeran, Kiribati with villages and main landmarks](https://github.com/zoccoler/blog/tree/napari_plugin_desing/marcelo_zoccoler/entry_user_interf3/images/21_Map_of_Tabuaeran_Kiribati_blue.png) by Government of USA, Government of Kiribati, under license CC BY-SA 3.0 <https://creativecommons.org/licenses/by-sa/3.0>, via Wikimedia Commons, where the following changes were made: green and red channels removed, then image was inverted regarding intensity.
+Put the image in the same folder as the code and let's expand it a bit:
+
+```
```

Comment:
```suggestion
```Python
```

Reply:
Adding `Python` here will color code according to Python-syntax and make it much more readable :)

Reply:
I'll add the same to the other code examples so you can merge it as a single change.